client/src/extension.ts

```diff
@@ -105,8 +105,12 @@ export async function activate (context: vscode.ExtensionContext, launchOptions:
   // Register commands for command palette
   registerCommand('flix.internalRestart', makeHandleRestartClient(context, { shouldUpdateFlix: false }))
   registerCommand('flix.internalDownloadLatest', makeHandleRestartClient(context, { shouldUpdateFlix: true }))
-
-  registerCommand('flix.cmdRunMain', handlers.makeHandleRunJobWithProgress(client, outputChannel, jobs.Request.cmdRunMain, 'Running Main'))
+  registerCommand('flix.cmdRunMain', handlers.RunInExistingTerminalWithoutArg)
+  registerCommand('flix.runMainWithArgs', handlers.RunInExistingTerminalWithArg)
+  registerCommand('flix.runMainNewTerminal', handlers.RunInNewTerminalWithoutArg)
+  registerCommand('flix.runMainNewTerminalWithArgs', handlers.RunInNewTerminalWithArg)
```

Comment:
Let's keep the names consistent, i.e.:

RunInExistingTerminalWithoutArg -> runMain
RunInExistingTerminalWithArg -> runMainWithArgs
RunInNewTerminalWithoutArg -> runMainNewTerminal
RunInNewTerminalWithArg -> runMainNewTerminalWithArgs

Reply:
Just noticed that you need `cmd` in front too!